sycl/test/CMakeLists.txt

```diff
@@ -37,7 +37,7 @@ configure_lit_site_cfg(
   ${CMAKE_CURRENT_SOURCE_DIR}/deploy-lit.site.cfg.py.in
   ${CMAKE_CURRENT_BINARY_DIR}/deploy-lit.site.cfg.py
   MAIN_CONFIG
-  ${CMAKE_CURRENT_SOURCE_DIR}/lit.cfg.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/deploy-lit.cfg.py
```

Comment:
I really don't like the idea that we had to copy this file and sync it in order to get all fixes/improvements.

I suggest to perform the same trick which were used to launch LIT tests in-tree back in 2017: https://github.com/llvm/llvm-project/commit/ce92db13ead9a24c767d60375545fa6ad6cb8ce0#diff-707931295fba376ca52d0861200e7adaL25.

In particular:
- re-order `configure_lit_site_cfg` directives, so, by default mapping is set for regular LIT tests
- add `--override-site-cfg` option or something like this which will allow to override default mapping `lit.cfg` -> `lit.site.cfg` for `check-sycl-deploy`

Reply:
I used it to remove deploy-lit.cfg.py at all. Take a look.